internal/command/group.go

```diff
@@ -0,0 +1,416 @@
+	}
+
+	groupAgg := GroupAggregateFromWriteModel(&existingGroup.WriteModel)
+	changedEvent, hasChanged, err := existingGroup.NewChangedEvent(
+		ctx,
+		groupAgg,
+		groupChange.Name,
+		groupChange.Description)
+	if err != nil {
+		return nil, err
+	}
+	if !hasChanged {
+		return nil, zerrors.ThrowPreconditionFailed(nil, "COMMAND-2M0fs", "Errors.NoChangesFound")
+	}
+	pushedEvents, err := c.eventstore.Push(ctx, changedEvent)
+	if err != nil {
+		return nil, err
+	}
+	err = AppendAndReduce(existingGroup, pushedEvents...)
+	if err != nil {
+		return nil, err
+	}
+	return groupWriteModelToGroup(existingGroup), nil
+}
+
+func (c *Commands) DeactivateGroup(ctx context.Context, groupID string, resourceOwner string) (*domain.ObjectDetails, error) {
+	if groupID == "" || resourceOwner == "" {
+		return nil, zerrors.ThrowInvalidArgument(nil, "COMMAND-98iF1", "Errors.Group.GroupIDMissing")
+	}
+
+	groupAgg, state, err := c.groupAggregateByID(ctx, groupID, resourceOwner)
+	if err != nil {
+		return nil, err
+	}
+
+	if state == domain.GroupStateUnspecified || state == domain.GroupStateRemoved {
+		return nil, zerrors.ThrowNotFound(nil, "COMMAND-223N9", "Errors.Group.NotFound")
+	}
+	if state != domain.GroupStateActive {
+		return nil, zerrors.ThrowPreconditionFailed(nil, "COMMAND-nle66", "Errors.Group.NotActive")
+	}
+
+	existingGroup, err := c.getGroupWriteModelByID(ctx, groupID, resourceOwner)
+	if err != nil {
+		return nil, err
+	}
+	if existingGroup.State == domain.GroupStateUnspecified || existingGroup.State == domain.GroupStateRemoved {
+		return nil, zerrors.ThrowNotFound(nil, "COMMAND-223N9", "Errors.Group.NotFound")
+	}
+	if existingGroup.State != domain.GroupStateActive {
+		return nil, zerrors.ThrowPreconditionFailed(nil, "COMMAND-nlf66", "Errors.Group.NotActive")
+	}
+
+	pushedEvents, err := c.eventstore.Push(ctx, group.NewGroupDeactivatedEvent(ctx, groupAgg))
+	if err != nil {
+		return nil, err
+	}
+
+	return &domain.ObjectDetails{
+		ResourceOwner: pushedEvents[0].Aggregate().ResourceOwner,
+		Sequence:      pushedEvents[0].Sequence(),
+		EventDate:     pushedEvents[0].CreatedAt(),
+	}, nil
+}
+
+func (c *Commands) ReactivateGroup(ctx context.Context, groupID string, resourceOwner string) (*domain.ObjectDetails, error) {
+	if groupID == "" || resourceOwner == "" {
+		return nil, zerrors.ThrowInvalidArgument(nil, "COMMAND-4fjsG", "Errors.Group.GroupIDMissing")
+	}
+
+	groupAgg, state, err := c.groupAggregateByID(ctx, groupID, resourceOwner)
+	if err != nil {
+		return nil, err
+	}
+
+	if state == domain.GroupStateUnspecified || state == domain.GroupStateRemoved {
+		return nil, zerrors.ThrowNotFound(nil, "COMMAND-4N9sd", "Errors.Group.NotFound")
+	}
+
+	if state != domain.GroupStateInactive {
+		return nil, zerrors.ThrowPreconditionFailed(nil, "COMMAND-6n8cs", "Errors.Group.NotInactive")
+	}
+
+	existingGroup, err := c.getGroupWriteModelByID(ctx, groupID, resourceOwner)
```

Comment:
Why is the group queried 2-times? Once with groupAggregateByID and once with getGroupWriteModelByID?